src/hotspot/os/posix/vmError_posix.cpp

```diff
@@ -76,22 +75,19 @@ static void crash_handler(int sig, siginfo_t* info, void* ucVoid) {
     pc = (address) info->si_addr;
   }
 
-  // Needed to make it possible to call SafeFetch.. APIs in error handling.
-  if (sig == SIGSEGV || sig == SIGBUS) {
-    if (uc && pc && StubRoutines::is_safefetch_fault(pc)) {
-      os::Posix::ucontext_set_pc(uc, StubRoutines::continuation_for_safefetch_fault(pc));
-      return;
-    }
+  // Handle safefetch here too, to be able to use SafeFetch() inside the error handler
+  if (handle_safefetch(sig, pc, uc)) {
+    return;
+  }
 
-    // Needed because asserts may happen in error handling too.
+  // Needed because asserts may happen in error handling too.
 #ifdef CAN_SHOW_REGISTERS_ON_ASSERT
-    if (info != NULL && info->si_addr == g_assert_poison) {
-      if (handle_assert_poison_fault(ucVoid, info->si_addr)) {
-        return;
-      }
+  if ((sig == SIGSEGV || sig == SIGBUS) && info != NULL && info->si_addr == g_assert_poison) {
+    if (handle_assert_poison_fault(ucVoid, info->si_addr)) {
+      return;
     }
-#endif // CAN_SHOW_REGISTERS_ON_ASSERT
   }
+#endif // CAN_SHOW_REGISTERS_ON_ASSERT
```

Comment:
Above hunks are not part of the patch, and should not be needed for safefetch. Why are they here?

Reply:
It's here to keep existing logic of this hunk. And to sync with mainline JDK version. The file was previously modified in scope of JDK-8289350 that doesn't affect JDK19+.

Reply:
Okay, I see now.